packages/fx-core/src/component/driver/middleware/addSWADeployTelemetry.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { HookContext, Middleware, NextFunction } from "@feathersjs/hooks";
+import { WrapDriverContext } from "../util/wrapUtil";
+import {
+  TeamsFxTelemetryConfig,
+  TeamsFxTelemetryReporter,
+} from "../../utils/teamsFxTelemetryReporter";
+import { TelemetryConstant } from "../../constant/commonConstant";
+import { performance } from "perf_hooks";
+import { TelemetryConstants } from "../../constants";
+import { isExecutionResult } from "./addStartAndEndTelemetry";
+import { maskSecretValues } from "../../utils/envUtil";
+
+/**
+ * A special telemetry middleware for SWA deployment.
+ * @param eventName The event name to be sent.
+ */
+export function addSWADeployTelemetry(eventName: string): Middleware {
+  return async (ctx: HookContext, next: NextFunction) => {
+    const name = ctx.arguments[4] as string | undefined;
+    const command = maskSecretValues(ctx.arguments[0].args as string);
+    // only add telemetry for script
+    if (name?.indexOf("deploy to Azure Static Web Apps") ?? -1 < 0) {
+      await next();
+      return;
+    }
+    const driverContext = ctx.arguments[1] as WrapDriverContext;
+    let telemetryReporter: TeamsFxTelemetryReporter | undefined = undefined;
+    if (driverContext.telemetryReporter) {
+      telemetryReporter = new TeamsFxTelemetryReporter(driverContext.telemetryReporter, {
+        componentName: TelemetryConstant.DEPLOY_TO_SWA_COMPONENT,
+      });
+    }
+    telemetryReporter?.sendStartEvent({ eventName });
```

Comment:
This telemetry was added in the NPX driver. It will act like the default telemetry but only works when name equals the SWA deploy script's default name.
The action of the GetDeploymentTokenFromAzureStaticWebApps has no additional telemetry.